### ISSUE TITLE
RandomCrop with padding and p<1 shifts keypoints and boxes on rows where the crop was not applied

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4473]
`RandomCrop` with `padding` set and `p < 1` shifts the keypoints and boxes of *unapplied* rows by the padding offset while returning those images untouched (main `5be4852b`, torch 2.14.0, CPU):

```python
import torch
from kornia.augmentation import AugmentationSequential, RandomCrop
x = torch.zeros(1, 1, 6, 10)
kp = torch.tensor([[[3.0, 2.0]]])
seq = AugmentationSequential(RandomCrop((4, 8), padding=(1, 2), p=0.0), data_keys=["input", "keypoints"])
out = seq(x, kp)
torch.equal(out[0], x)   # True  -- image not cropped, not padded
out[1]                   # tensor([[[4., 4.]]])  -- keypoint moved by (+1, +2)
```

The image side of `RandomCrop` applies the pad only on rows where `batch_prob` fires, but `apply_transform_keypoint` (`kornia/augmentation/_2d/geometric/crop.py:191`) pads unconditionally, so on an unapplied row the label no longer corresponds to the pixels. The same offset reaches boxes. `p=1.0` is consistent; the bug is only visible with `p < 1` (or a batch where some rows are skipped).

**Expected:** the keypoint/box pad translation is applied only on rows where the crop was applied, mirroring the image path, so a label round-trip through `inverse` (or an identity check on unapplied rows) holds.

Found by the geometry audit in #4465, whose `RandomCrop(p=0.5)` round-trip error of `sqrt(5)` on unapplied rows is exactly this `(1, 2)` offset; the audit follows the pixels, which is the right contract, so this is a container defect rather than an audit defect.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
